### DISCUSSION TITLE
Why can't I use `atlas migrate diff` to generate a new migration file after using the `atlas migrate apply` command?

### DISCUSSION THREAD

**Discussion by rehth** [ariga/atlas#1995] [Q&A] [answered]
OS: Windows 11
Version: 0.13.1

Error: sql/migrate: taking database snapshot: sql/migrate: connected database is not clean: found table "atlas_schema_revisions" in connected schema

**Comment by a8m** [comment, marked as answer]
Hey @rehth 👋 

The concept of the dev-database is explained here: https://atlasgo.io/atlas-schema/sql#dev-database. In short, you cannot use the same database for both `apply` and `diff`. The first represents the "real" database (or is the actual one), while the second is used in development to validate the migration directory and compute the new SQL diff.

> As mentioned above, Atlas uses the dev database to compute the desired state of the database schema. Therefore, before starting its work, Atlas ensures the dev database is clean and there are no leftovers from previous runs. Once done, Atlas cleans up after itself and the dev database is ready for future runs.
